src/WorkOS.net/Services/Webhooks/WebhookService.cs

```diff
@@ -0,0 +1,219 @@
+        /// time tolerance for timing attacks.
+        /// </param>
+        /// <returns>Boolean True or False of verification.</returns>
+        public bool VerifyHeader(string payload, string sigHeader, string secret, long tolerance)
+        {
+            var timeAndSignature = this.GetTimestampAndSignature(sigHeader);
+            var timeStamp = timeAndSignature.Item1;
+
+            if (!this.Verify_Time_Tolerance(timeStamp, tolerance))
+            {
+                throw new Exception("Timestamp outside of the tolerance zone");
+            }
+
+            string signatureHash = timeAndSignature.Item2;
+            string expectedSig = this.ComputeSignature(timeStamp, payload, secret);
+
+            if (this.SecureCompare(expectedSig, signatureHash))
+            {
+                return true;
+            }
+            else
+            {
+                throw new Exception("Signature hash does not match the expected signature hash for payload");
+            }
+        }
+
+        /// <summary>
+        /// Get timestamp and signature hash.
+        /// </summary>
+        /// <param name="signatureHeader">
+        /// Signatures header.
+        /// </param>
+        /// <returns> Tuple of [DateTime timestamp, string signaturehash].</returns>
+        public (string Timestamp, string SignatureHash) GetTimestampAndSignature(string signatureHeader)
+        {
+            var timeAndSig = signatureHeader.Split(',');
+            var timeStamp = timeAndSig[0];
+            var signatureHash = timeAndSig[1];
+            if (string.IsNullOrEmpty(timeStamp) || string.IsNullOrEmpty(signatureHash))
+            {
+                throw new ArgumentException("Unable to extract timestamp and signature hash from header");
+            }
+
+            var removeT = "t=";
+            var t = timeStamp.IndexOf(removeT);
+            if (t != -1)
+            {
+                timeStamp = timeStamp.Substring(t + removeT.Length);
+            }
+
+            if (string.IsNullOrEmpty(timeStamp))
+            {
+                throw new Exception("Unable to extract timestamp");
+            }
+
+            var removeV = "v1=";
+            var v = signatureHash.IndexOf(removeV);
+            if (v != -1)
+            {
+                signatureHash = signatureHash.Substring(v + removeV.Length, 64);
+            }
+
+            if (string.IsNullOrEmpty(signatureHash))
+            {
+                throw new Exception("Unable to extract signatureHash");
+            }
+
+            return (Timestamp: timeStamp, SignatureHash: signatureHash);
+        }
+
+        /// <summary>
+        /// Convert Unix time value to a DateTime object.
+        /// </summary>
+        /// <param name="unixtime">The Unix time stamp you want to convert to DateTime.</param>
+        /// <returns>Returns a DateTime object that represents value of the Unix time.</returns>
+        public DateTime UnixTimeToDateTime(long unixtime)
```

Comment:
Does this helper method need to be exposed outside of the class?

We should limit the visibility of methods that aren't a part of this class' public API.

Reply:
It does not, changed to private